test/end2end_test.go

```diff
@@ -2124,3 +2127,25 @@ func (fw *filterWriter) Write(p []byte) (n int, err error) {
 	}
 	return fw.dst.Write(p)
 }
+
+func equalErrors(l, r error) bool {
+	if l == nil && r != nil {
+		return false
+	}
+	if l != nil && r == nil {
+		return false
+	}
+	return grpc.Code(l) == grpc.Code(r) && grpc.ErrorDesc(l) == grpc.ErrorDesc(r)
+}
```

Comment:
this func can be simply

`return grpc.Code(l) == grpc.Code(r) && grpc.ErrorDesc(l) == grpc.ErrorDesc(r)``